src/Controller/BlogPostController.php

```diff
@@ -0,0 +1,207 @@
+
+        $posts_json = json_encode(
+            array_map(array($blogPostService, 'getBlogPostReturnData'), $posts),
+            JSON_THROW_ON_ERROR
+        );
+
+        return new Response($posts_json, 200, ['Content-Type' => 'application/json']);
+    }
+
+    /**
+     * @Route("/blogposts/{userID}/create", name="blog-posts-create", methods={"POST"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     */
+    public function createBlogPost(
+        Request $request,
+        EntityManagerInterface $entityManager,
+        DTOSerializer $serializer,
+        int $userID,
+    ):
+    Response {
+        /** @var BlogPostEnquiry $postEnquiry */
+        $postEnquiry = $serializer->deserialize(
+            $request->getContent(),
+            BlogPostEnquiry::class,
+            'json'
+        );
+
+        /** @var BlogUserRepository $blogPostRepository */
+        $blogUserRepository = $entityManager->getRepository(BlogUser::class);
+        $user = $blogUserRepository->find($userID);
+
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+        $blogPostService = new BlogPostService($blogPostRepository);
+
+        if (!$user) {
+            throw $this->createNotFoundException(
+                'No user found for id' . $userID
+            );
+        }
+
+        if (empty($postEnquiry->getBody())) {
+            return new Response(
+                json_encode(["success" => false, "msg" => "Missing body text"], JSON_THROW_ON_ERROR),
+                400,
+                ['Content-Type' => 'application/json']
+            );
+        }
+
+        if ($postEnquiry->getTimestamp() === null) {
+            return new Response(
+                json_encode(["success" => false, "msg" => "Missing timestamp text"], JSON_THROW_ON_ERROR),
+                400,
+                ['Content-Type' => 'application/json']
+            );
+        }
+
+        $post = $blogPostService->createBlogPostFromEnquiry($postEnquiry, $user);
+
+
+        $entityManager->persist($post);
+        $entityManager->flush();
+
+        return new Response(
+            json_encode(["success" => true], JSON_THROW_ON_ERROR),
+            200,
+            ['Content-Type' => 'application/json']
+        );
+    }
+
+
+    /**
+     * @Route("/blogposts/{blogID}", name="blog-posts-delete", methods={"DELETE"})
+     * @throws NotSupported
+     * @throws \JsonException
+     *
+     */
+    public function removeBlogPost(
+        EntityManagerInterface $entityManager,
+        int $blogID,
+    ):
+    Response {
+
+        /** @var BlogPostRepository $blogPostRepository */
+        $blogPostRepository = $entityManager->getRepository(BlogPost::class);
+
+        $post = $blogPostRepository->find($blogID);
+
+        if (!$post) {
+            throw $this->createNotFoundException(
+                'No post found for id' . $blogID
+            );
+        }
+
```

Comment:
[Tony] should be in repository (mentioned by Nick before)
[Jeff] It's okay to throw exceptions from repository class